homeassistant/components/notify/nma.py

```diff
@@ -0,0 +1,98 @@
+"""
+homeassistant.components.notify.nma
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+NMA (Notify My Android) notification service.
+
+Configuration:
+
+To use the NMA notifier you will need to add something like the following
+to your config/configuration.yaml
+
+notify:
+  platform: nma
+  api_key: YOUR_API_KEY
+
+VARIABLES:
+
+api_key
+*Required
+Enter the API key for NMA. Go to https://www.notifymyandroid.com and create a
+new API key to use with Home Assistant.
+
+Details for the API : https://www.notifymyandroid.com/api.jsp
+
+"""
+import logging
+import xml.etree.ElementTree as ET
+
+from homeassistant.helpers import validate_config
+from homeassistant.components.notify import (
+    DOMAIN, ATTR_TITLE, BaseNotificationService)
+from homeassistant.const import CONF_API_KEY
+
+_LOGGER = logging.getLogger(__name__)
+_RESOURCE = 'https://www.notifymyandroid.com/publicapi/'
+
+
+def get_service(hass, config):
+    """ Get the NMA notification service. """
+
+    if not validate_config(config,
+                           {DOMAIN: [CONF_API_KEY]},
+                           _LOGGER):
+        return None
+
+    try:
+        # pylint: disable=unused-variable
+        from requests import Request, Session
+
+    except ImportError:
+        _LOGGER.exception(
+            "Unable to import requests. "
+            "Did you maybe not install the 'Requests' package?")
+
+        return None
+
+    nma = Session()
+    response = nma.get(_RESOURCE + 'verify',
+                       params={"apikey" : config[DOMAIN][CONF_API_KEY]})
+    tree = ET.fromstring(response.content)
```

Comment:
The return code for the API call is embedded in the XML response.

Success: 
```
<?xml version="1.0" encoding="UTF-8"?><nma><success code="200" remaining="995" resettimer="51"/></nma>
```
Failure: 
```
<?xml version="1.0" encoding="UTF-8"?><nma><error code="400" >Parameter \'apikey\' length is invalid.</error></nma>
```
The HTTP response code is for both 200 because they were successful. With the parsing of the XML it's possible to display an advanced error message in the log.